### ISSUE TITLE
invalid or missing instance

### ISSUE THREAD

**Issue by paullaffitte** [rollbar/rollbar-php#265]
I'm upgrading a project from 0.18.2 to 1.3.1 and it's working a few times until Rollbar::log start to return an error "invalid or missing instance", with a 400 staus. Then, it continue to return the same error and doesn't log nothing.
I saw a similar issue (#237), but it doesn't help me to solve this.

If it can be useful, here is a var_dump of my $_SESSION just before I try to log
```
array (size=6)
  'useragent6' => 
    array (size=16)
      'browser' => string 'Chrome' (length=6)
      'browser_maker' => string 'Google Inc' (length=10)
      'comment' => string 'Chrome 61.0' (length=11)
      'crawler' => boolean false
      'device_pointing_method' => string 'mouse' (length=5)
      'device_type' => string 'Desktop' (length=7)
      'majorver' => int 61
      'minorver' => int 0
      'parent' => string 'Chrome 61.0' (length=11)
      'platform' => string 'Linux' (length=5)
      'version' => float 61
      'browser_name_pattern' => string 'Mozilla/5.0 (*Linux*) applewebkit* (*khtml*like*gecko*) Chrome/61.0*Safari/*' (length=76)
      'browser_name_regex' => string '/^Mozilla\/5\.0 \(.*Linux.*\) applewebkit.* \(.*khtml.*like.*gecko.*\) Chrome\/61\.0.*Safari\/.*$/' (length=98)
      'ismobiledevice' => boolean false
      'istablet' => boolean false
      'useragent' => string 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.79 Safari/537.36' (length=104)
  'ip_b' => string '��' (length=4)
  'fingerprint2_b' => string '�H~b�lɬ����' (length=16)
  'user_fingerprint_id' => string '89' (length=2)
  'subscribed' => boolean true
  'userData' => 
    array (size=2)
      '_ip' => string '172.18.0.1' (length=10)
      'data' => 
        array (size=3)
          'Username' => string 'Guess' (length=5)
          'ID' => int 0
          'Password' => int 0
```
Edit:
I had a problem also with the previous version we were using (0.18.2). At least the function report_message was triggering a notice "iconv(): Detected an incomplete multibyte character in input string".
Now I "solved" both issues by clearing the cache of my browser.

**Comment by rokob** [comment]
That error implies that whatever payload we send is missing the `data` key. So there must be a bug in how we are building the payload or doing scrubbing. Let me try to look into this a little further.

**Comment by rokob** [comment]
Okay what I think is happening is that either during the building of data or during scrubbing (probably during scrubbing), the data value is getting set to null for some reason. We then subsequently don't include it in the final payload because of the way `serializeForRollbar` works. I'm trying to figure out how data could end up null.

**Comment by paullaffitte** [comment]
I upgraded to the latest version today again (1.3.6), I keep have the same issue. Did you find something about it ? Do you need anything that could help you figuring out where the bug comes from ?

Thank you

**Comment by ArturMoczulski** [comment]
@paullaffitte Do you think you can do some debugging in your Rollbar package? I suggest adding `var_dump` or your other preferred debugging tool in `\Rollbar\RollbarLogger::log` method on line 69 of file `src/RollbarLogger.php` to inspect all of the subsequent paylods:

```php
var_dump($toSend);
```

Thanks to this we should be able to see what and when starts disappearing from your payload.

**Comment by rivkahstandig3636** [comment]
@paullaffitte Is this still an issue for you?

**Comment by h0gar** [comment]
Having the same issue. Rollbar returns "invalid or missing instance" with status 400.

In my case, after digging into the rollbar library, I found out the error comes from one of the frame's arguments. Maybe because some arguments are too large. Replacing '"args" => $this->args,' by '"args" => [],' in the file Payload/Frame.php fixes it for me. But then I lose the arguments in the report's trace.

I figured a way to avoid these corrupting arguments and keep others with:

```
        $args = $this->args;
        if(!json_encode($args))
            $args = [];
```

Not sure though if works for all corrupting arguments but in my case it seems to do the job.

**Comment by ArturMoczulski** [comment]
@h0gar can you provide us with a dump of arguments that are causing the error? I need to build tests for this

**Comment by ArturMoczulski** [comment]
@paullaffitte PR 332 should fix this issue. Can you please check with current `master` or tag `v1.4.1` when it gets released?

**Comment by abuisine** [comment]
I do confirm that the problem is solved in 1.4.3:
binary values stored in session are now displayed as None on rollbar.com rather than triggering a 400 error code 'invalid or missing instance' on the php side.
